src/yarp.c

```diff
@@ -8343,13 +8343,14 @@ parse_starred_expression(yp_parser_t *parser, yp_binding_power_t binding_power,
 }
 
 // Convert the name of a method into the corresponding write method name. For
-// exmaple, foo would be turned into foo=.
+// example, foo would be turned into foo=.
 static void
-parse_write_name(yp_string_t *string) {
+parse_write_name(yp_string_t *string) { // TODO should change (yp_constant_id_t *string)
     // The method name needs to change. If we previously had
     // foo, we now need foo=. In this case we'll allocate a new
     // owned string, copy the previous method name in, and
     // append an =.
+    // TODO how to get the string from a yp_constant_id_t* ?
```

Comment:
I think we'll need https://github.com/ruby/yarp/pull/1449#issuecomment-1716031475 or similar

Reply:
You would go to the constant pool and pull out the `yp_constant_t`, it will have the source and length on it.

Reply:
> pull out the `yp_constant_t`

How to do that?
I didn't consider iterating the whole `yp_constant_pool_t->constants` because that would be linear and likely way too slow.

Reply:
How does this work for CRuby compilation? There must be a way to convert a `yp_constant_id_t` to a `char*` or `ID`, right?

Reply:
CRuby pre-fills an array: https://github.com/ruby/ruby/blob/master/compile.c#L874-L883

I was going to rearrange the constant pool so that you don't have to do that, and can index directly into the pool to acquire the `char*`.

Reply:
Thanks for the pointer. That seems suboptimal, especially if there are many empty slots and of course #slots > #constants. And we can't really use that approach in this code path.

Maybe I can workaround for now using `call->message_loc`, since that seems used just before `parse_write_name()` is called anyway (`call->message_loc` is not correct for `foo[arg] = val` but probably that's not used in those places?).

> I was going to rearrange the constant pool so that you don't have to do that, and can index directly into the pool to acquire the char*.

Yes, I think this is generally useful and convenient.

Reply:
I have updated this part to use `pm_constant_pool_t->constants`.